.github/workflows/yocto-build-deploy.yml

```diff
@@ -0,0 +1,1174 @@
+name: "Yocto Build-Test-Deploy"
+
+on:
+  workflow_call:
+    secrets:
+      # API key names should be descriptive telling us what company, what for, where from, what's the key.
+      # Let's say, XCompany wants to have test, then keys could be: XCOMPANY_AUTOKIT_BM_ACCESS_TOKEN
+      AUTOKIT_BALENACLOUD_API_KEY:
+        description: balena API key for Leviathan testing from BalenaCloud
+        required: false
+      AUTOKIT_BALENAMACHINE_API_KEY:
+        description: balena API key for Leviathan testing from balena-os BM instance
+        required: false
+      BALENA_API_KEY:
+        description: balena API key for deploying releases # TODO: Different for staging and prod - add another Secret for staging key
+        required: false
+      BALENAOS_STAGING_TOKEN:
+        description: balena API key for deploying releases to staging # TODO: Different for staging and prod - add another Secret for staging key
+        required: false
+      DOCKERHUB_USER:
+        description: Dockerhub user for pulling private helper images
+        required: false
+      DOCKERHUB_TOKEN:
+        description: Dockerhub token for pulling private helper images
+        required: false
```

Comment:
We can remove these if we stop using the balena-img private helper image for prepare.ts

Reply:
added as comments in the workflow file